test/cli/fixtures/each-array-labels.js

```diff
@@ -0,0 +1,43 @@
+// Automatic labels for test.each() array data where possible
+// https://github.com/qunitjs/qunit/issues/1733
+
+QUnit.test.each('array of arrays', [[1, 2, 3], [1, 1, 2]], function (assert, _data) {
+  assert.true(true);
+});
+
+QUnit.test.each('array of simple strings', [
+  'foo',
+  'x'.repeat(40),
+  '$',
+  'http://example.org',
+  ' ',
+  ''
+], function (assert, _data) {
+  assert.true(true);
+});
+
+QUnit.test.each('array of mixed', [
+  undefined,
+  null,
+  false,
+  true,
+  0,
+  1,
+  -10,
+  10 / 3,
+  10e42,
+  Infinity,
+  NaN,
+  [],
+  {},
+  '999: example',
```

Comment:
maybe we could also add a simple string here, to make it clear via the test assertion that even for mixed arrays, a simple string will not be prefixed with the index
```suggestion
  '999: example',
  'simple string'
```

Reply:
Good idea, yeah. I hope it'll be rare for a test to contain both, but it is what it is.